website/blog/2025-04-18-replacing-mongodb-with-ferretdb-librechat.md

```diff
@@ -0,0 +1,112 @@
+---
+slug: replacing-mongodb-with-ferretdb-librechat
+title: 'Replacing MongoDB with FerretDB in LibreChat'
+authors: [alex]
+description: >
+  A guide to replacing MongoDB with FerretDB for a fully open-source LibreChat setup.
+image: /img/blog/ferretdb-librechat.jpg
+tags:
+  [
+    tutorial,
+    document databases,
+    community,
+    open source,
+    compatible applications
+  ]
+---
+
+![Replacing MongoDB with FerretDB on LibreChat](/img/blog/ferretdb-librechat.jpg)
+
+[LibreChat](https://www.librechat.ai/) is a free, open-source application that provides a user-friendly and customizable interface for interacting with various AI providers.
+It allows users to connect with providers like [OpenAI](https://openai.com/), [Azure](https://azure.microsoft.com/), [Anthropic](https://www.anthropic.com/), and more.
+
+For LibreChat users who want to stay fully open source, FerretDB is a great drop-in replacement for MongoDB,
+especially if you're looking to avoid proprietary databases or vendor lock-in.
+It uses PostgreSQL with DocumentDB extension as the backend, while letting you use familiar MongoDB operations and commands.
+
+This guide shows how to run LibreChat with FerretDB as the database, either by connecting to an existing FerretDB instance or running everything in Docker.
+
+<!--truncate-->
+
+## How to use FerretDB with LibreChat
+
+LibreChat can use FerretDB as its database in two ways:
+
+### Option 1: Connect to an existing FerretDB instance
+
+If you already have FerretDB running, simply swap out MongoDB connection string with your FerretDB connection string using the `MONGO_URI` environment variable.
```

Comment:
Maybe having a prerequisite step with link to how to setup LibreChat might be useful :hugs: